lib/room.cpp

```diff
@@ -1506,12 +1508,17 @@ void Room::handleRoomKeyEvent(const RoomKeyEvent& roomKeyEvent,
                                   roomKeyEvent.sessionKey())) {
         qCWarning(E2EE) << "added new inboundGroupSession:"
                       << d->groupSessions.size();
-        for (unsigned long int i = 0; i < d->timeline.size(); i++) {
-            if (auto encryptedEvent = d->timeline[i].viewAs<EncryptedEvent>()) {
+        for (const auto& eventId : d->undecryptedEvents[roomKeyEvent.sessionId()]) {
+            if (!d->eventsIndex.contains(eventId)) {
+                continue;
+            }
+            auto event = d->timeline.rend() - (d->eventsIndex.value(eventId) - minTimelineIndex() + 1);
+            if (auto encryptedEvent = event->viewAs<EncryptedEvent>()) {
```

Comment:
Copy-pasted from `Room::Private::processReplacement` in `room.cpp`. Might make sense to make it more generic so that it can be reused both for edits and decryption. I can FTFY if you struggle.
```suggestion
            const auto pIdx = d->eventsIndex.constFind(eventId);
            if (pIdx == d->eventsIndex.cend())
                continue;
            auto& ti = d->timeline[Timeline::size_type(*pIdx - minTimelineIndex())];
            if (auto encryptedEvent = ti.viewAs<EncryptedEvent>()) {
```

Reply:
If you want to do that, please do - it's definitely going to be faster than if i do it